### PR TITLE
Optimizations and features updates

### PR DESCRIPTION
**Pull request by proy9714**
1. Speed up by using async sub-query processing
2. Support for saving report in docx format
3. Support for python 3.10
4. Updated imports as per latest langchain standards
5. Replaced deprecated lc_openai in favour of more modern ChatOpenAI langchain approaches

### CODE REVIEW

**Comment by proy9714** [comment]
Hey @assafelovic a quick review on this PR?!

**Review by assafelovic — APPROVED** [review]

**Comment by assafelovic** [comment]
Thanks for your amazing contributions @proy9714  ! Can you also please update requirements.txt?

**Comment by assafelovic** [comment]
Also @proy9714 would be important to add in UI to select the proper download format. What are your thoughts on how to use it or choose between pdf and word?

File: backend/utils.py [hunk 1]
```diff
@@ -42,3 +45,31 @@ async def write_md_to_pdf(text: str) -> str:
 
     encoded_file_path = urllib.parse.quote(f"{file_path}.pdf")
     return encoded_file_path
+
+async def write_md_to_word(text: str) -> str:
```
Comment (assafelovic) [inline backend/utils.py:49]: How are you planning to use it? Should we maybe add a config flag that if True will also download a docx version? Or have it in the UI for selection? @proy9714

File: pyproject.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@ license = "MIT"
 readme = "README.md"
```
Comment (assafelovic) [inline pyproject.toml:7]: @proy9714 can you also please update requirements.txt file?

**Review by assafelovic — CHANGES_REQUESTED** [review]
Add inline comments for review

File: backend/utils.py (hunk 1 above)
Comment (proy9714) [reply backend/utils.py:49]: I was hoping someone could contribute and make a UI button that downloads the word document if clicked.

File: pyproject.toml (hunk 2 above)
Comment (proy9714) [reply pyproject.toml:7]: Ok will do

**Comment by proy9714** [comment]
Hey @assafelovic added requested changes!

**Comment by proy9714** [comment]
Hey @assafelovic added requested changes! Also a UI button for downloading the report in docx!

File: requirements.txt [hunk 3]
```diff
@@ -1,27 +1,113 @@
-# dependencies
```
Comment (assafelovic) [inline requirements.txt:1]: @proy9714 hey sorry to nag no need for this conditional file. Simply add the new packages without their version. After that happy to merge, thanks! 👯

**Comment by proy9714** [comment]
@assafelovic Done!

**Review by assafelovic — APPROVED** [review]
Perfect, thank you!

**Comment by assafelovic** [comment]
Super appreciate your contributions @proy9714 👏

**Comment by badcom** [comment]
Thanks for this, @proy9714 

Clicking the "Download as Docx" button refreshed the page and didn't trigger the file download =/

**Comment by proy9714** [comment]
@badcom I think its not refreshing the page its opening a new tab and there should be a dialog box asking you to save the docx.....

**Pull request merged**